backend-rust/src/indexer.rs

```diff
@@ -3818,7 +3838,9 @@ impl PreparedSpecialTransactionOutcomeUpdate {
                     AccountStatementEntryType::FoundationReward,
                 )?;
 
-                Self::Payday(PreparedPayDayBlock::prepare(block_height, rewards)?)
+                Self::PreparedPayDayBlock(
```

Comment:
We are using the `Mint` special transaction outcome for signalling payday, but the `Mint` happened before P4 where payday got introduced, so maybe not the best place, but not sure if it is an issue either.
In the rust SDK we actually have a function checking for one of the events prefixed payday instead. https://docs.rs/concordium-rust-sdk/latest/src/concordium_rust_sdk/types/mod.rs.html#1025-1037